### ISSUE TITLE
Bug: `globalReturn` fails with `sourceType: "module"`

### ISSUE THREAD

**Issue by erosman** [eslint/eslint#20527]
### Environment

Node version: 25.6.0
npm version: 11.10.0
Local ESLint version: 10.0.0
Global ESLint version: 10.0.0
Operating System: Ubuntu 24.04

### What parser are you using?

Default (Espree)

### What did you do?

Setting `globalReturn` works with `sourceType`  `'commonjs'` or `'scripts'`, but does not work with  `"module"`.
Tested with ESLint 9 & 10.

```js
  parserOptions: {
    ecmaFeatures: {
      globalReturn: true,
    },
  },
```

See also: [Future of ecmaFeatures.globalReturn](https://github.com/eslint/js/issues/525)

### What did you expect to happen?

`globalReturn` to work with  `sourceType: "module"`

### What actually happened?

> Parsing error: 'return' outside of function

### Link to Minimal Reproducible Example

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLy8gcmV0dXJuIG91dCBvZiBmdW5jdGlvblxuaWYgKHRydWUpIHtcbiAgcmV0dXJuO1xufSIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnsiZ2xvYmFsUmV0dXJuIjp0cnVlfX19fX0=)

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
This is expected as we [set globalReturn to false for modules](https://github.com/eslint/eslint/blob/0bd54976080936ce080ee2552d504180105593b4/lib/languages/js/index.js#L147) (at least since ESLint v9).
I see this issue as a duplicate of the linked issue which is still open and planned for v11, which seems to rather add an additional property such as `wrapperFunction` than allowing `globalReturn` for modules.
